src/lib/decodeData.ts

```diff
@@ -139,27 +141,19 @@ export const decodeTupleKeyValue = (
 ): Array<string> => {
   // We assume data has already been validated at this stage
 
-  let valueTypeToDecode = valueType;
+  // Sanitize the string to keep only the tuple, if we are dealing with `CompactBytesArray`
+  const valueTypeToDecode = valueType.replace(COMPACT_BYTES_ARRAY_STRING, '');
 
-  if (valueType.includes('[CompactBytesArray')) {
-    valueTypeToDecode = valueType.replace(COMPACT_BYTES_ARRAY_STRING, '');
-  }
-
-  const valueTypeParts = valueTypeToDecode
-    .substring(1, valueTypeToDecode.length - 1)
-    .split(',');
-  const valueContentParts = valueContent
-    .substring(1, valueContent.length - 1)
-    .split(',');
+  const valueTypeParts = extractTupleElements(valueTypeToDecode);
+  const valueContentParts = extractTupleElements(valueContent);
 
   const bytesLengths: number[] = [];
+
   valueTypeParts.forEach((valueTypePart) => {
     const regexMatch = valueTypePart.match(tupleValueTypesRegex);
 
     // if we are dealing with `bytesN`
-    if (regexMatch) {
-      bytesLengths.push(parseInt(regexMatch[1], 10));
-    }
+    if (regexMatch) bytesLengths.push(parseInt(regexMatch[1], 10));
```

Comment:
personal preference - but i prefer with brackets, i think it is easier to read and see it is an if